js/fileUtils.js

```diff
@@ -1,6 +1,29 @@
 import invariant from "invariant";
+import readStream from "filereader-stream";
 
-export function genMediaTags(file) {
+import http from "stream-http";
+
+async function sourceToStream(source) {
+  if (typeof source === "string") {
+    // Assume URL
+    return new Promise(resolve => {
+      http.get(source, stream => {
+        resolve({
+          stream,
+          type: stream.headers["content-type"]
+        });
+      });
+    });
+  }
+  // Assume Blob
+  return {
+    stream: readStream(source),
+    type: source.name,
+    size: source.size
+  };
+}
+
+export async function genMediaTags(file, mm, observer) {
```

Comment:
`mm` is not a great variable name in my opinion. Maybe we could just pass in a `parseStream` function?